### ISSUE TITLE
Indented `{{> @partial-block }}` cannot handle a SafeString.

### ISSUE THREAD

**Issue by jbsulli** [handlebars-lang/handlebars.js#1695]
* [x] This is NOT a security issues!! If it is, please file a report at https://www.npmjs.com/advisories/report?package=handlebars**

# Bug description
Indented `{{> @partial-block }}` cannot handle a SafeString. I run into this when I have a block helper that returns an instance of `SafeString` as the only content in a partial block. Easiest to checkout the included example link.

# Example link

https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%7B%7B%23%3Emy_partial%7D%7D%7B%7B%23safe%7D%7D%7B%7B%2Fsafe%7D%7D%7B%7B%2Fmy_partial%7D%7D%5Cn%22%2C%22partials%22%3A%5B%7B%22name%22%3A%22my_partial%22%2C%22content%22%3A%22%20%5Cn%20%20%20%7B%7B%3E%20%40partial-block%20%7D%7D%22%7D%5D%2C%22input%22%3A%22%7B%7D%22%2C%22output%22%3A%22%20%5Cn%20%20%20%7B%7Btest%7D%7D%5Cn%22%2C%22preparationScript%22%3A%22Handlebars.registerHelper('safe'%2C%20function%20()%20%7B%5Cn%20%20%20%20return%20new%20Handlebars.SafeString(%5C%22%7B%7Btest%7D%7D%5C%22)%3B%5Cn%7D)%5Cn%22%2C%22handlebarsVersion%22%3A%224.7.6%22%7D

# Expected outcome

Renders an indented, safe string.

# Actual outcome

Throws a `.split is not a function` error.

**Comment by jbsulli** [comment]
A hack that works for me right now is to include a comment in front of the `{{> @partial-block }}` like so:
```
  {{! this comment fixes https://github.com/handlebars-lang/handlebars.js/issues/1695 }}{{> @partial-block }}
```

Also, I noticed that this only happens for block helpers. [A normal helper doesn't have this issue](https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%7B%7B%23%3Emy_partial%7D%7D%7B%7Bsafe%7D%7D%7B%7B%2Fmy_partial%7D%7D%5Cn%22%2C%22partials%22%3A%5B%7B%22name%22%3A%22my_partial%22%2C%22content%22%3A%22%7B%7B!%20indented%20partial%20block%20%7D%7D%5Cn%20%20%7B%7B%3E%20%40partial-block%20%7D%7D%22%7D%5D%2C%22input%22%3A%22%7B%7D%22%2C%22output%22%3A%22%20%20%7B%7Btest%7D%7D%5Cn%5Cn%22%2C%22preparationScript%22%3A%22Handlebars.registerHelper('safe'%2C%20function%20()%20%7B%5Cn%20%20%20%20return%20new%20Handlebars.SafeString(%5C%22%7B%7Btest%7D%7D%5C%5Cn%5C%22)%3B%5Cn%7D)%5Cn%22%2C%22handlebarsVersion%22%3A%224.7.6%22%7D).

**Comment by jbsulli** [comment]
I noticed that this only happens for block helpers. [A normal helper doesn't have this issue](https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%7B%7B%23%3Emy_partial%7D%7D%7B%7Bsafe%7D%7D%7B%7B%2Fmy_partial%7D%7D%5Cn%22%2C%22partials%22%3A%5B%7B%22name%22%3A%22my_partial%22%2C%22content%22%3A%22%7B%7B!%20indented%20partial%20block%20%7D%7D%5Cn%20%20%7B%7B%3E%20%40partial-block%20%7D%7D%22%7D%5D%2C%22input%22%3A%22%7B%7D%22%2C%22output%22%3A%22%20%20%7B%7Btest%7D%7D%5Cn%5Cn%22%2C%22preparationScript%22%3A%22Handlebars.registerHelper('safe'%2C%20function%20()%20%7B%5Cn%20%20%20%20return%20new%20Handlebars.SafeString(%5C%22%7B%7Btest%7D%7D%5C%5Cn%5C%22)%3B%5Cn%7D)%5Cn%22%2C%22handlebarsVersion%22%3A%224.7.6%22%7D).

**Comment by imike57** [comment]
Any fix for this ? 
Would be nice to have HTML generated well indented when using Partial Blocks... Currently it's only indented when using Basics partials like `{{> myPartial}}`, 

but if I use 

```
  {{#> mypartial}}
    Some data ...
  {{/myPartial}}
```

It's not indented.

**Comment by xmedeko** [comment]
Yes, I have [same problem with inline partial](https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%7B%7B%23*inline%20%5C%22myPartial%5C%22%7D%7D%7B%7B%23bold%7D%7DHello%20%7B%7Bfirstname%7D%7D%7B%7B%2Fbold%7D%7D%7B%7B%2Finline%7D%7D%5Cn%7B%7B%23each%20people%7D%7D%5Cn%20%20%7B%7B%3E%20myPartial%7D%7D%5Cn%7B%7B%2Feach%7D%7D%22%2C%22partials%22%3A%5B%5D%2C%22input%22%3A%22%7B%5Cn%20%20people%3A%20%5B%5Cn%20%20%20%20%7B%20firstname%3A%20%5C%22Nils%5C%22%20%7D%2C%5Cn%20%20%20%20%7B%20firstname%3A%20%5C%22Yehuda%5C%22%20%7D%2C%5Cn%20%20%5D%2C%5Cn%7D%5Cn%22%2C%22output%22%3A%22%5Cn%20%20%20%3Cb%3EHello%20Nils%3C%2Fb%3E%20%20%20%3Cb%3EHello%20Yehuda%3C%2Fb%3E%22%2C%22preparationScript%22%3A%22Handlebars.registerHelper(%5C%22bold%5C%22%2C%20function(options)%20%7B%5Cn%20%20return%20new%20Handlebars.SafeString('%3Cb%3E'%20%2B%20options.fn(this)%20%2B%20%5C%22%3C%2Fb%3E%5C%22)%3B%5Cn%7D)%3B%22%2C%22handlebarsVersion%22%3A%224.7.7%22%7D)

@imike57 Try to add a space `{{#>my_partial}} {{#safe}}{{/safe}}{{/my_partial}}`

IMO the fix may be simple - convert SafeString from partial to the plain string by `"" + result`.
